encoder.py

```diff
@@ -48,7 +48,21 @@ def calc(MSG, hash_type):
 
         result = [str(bin(ord(elem))[2:]) for elem in MSG]
         return ' '.join(result)
-
+    
+     elif hash_type == 'book':
+        """Convert message to key representation"""
+        
+        passage = click.prompt("Enter passage", type=str)
+        w_l=passage.split(' ')
+        key=[]
+        f_l_l=[a[0].lower() for a in w_l]
+        for i in MSG.lower():
+            if i not in f_l_l:
+                raise ValueError("Error: The entered passage does not support the message.")
+            if i in f_l_l:
+                key.append(str(f_l_l.index(i)+1))
+        return ','.join(key)
```

Comment:
Correct me if I misunderstood something but shouldn't the result be separated by spaces instead of commas?

Reply:
Both ways seem fine. Irrespective, the change has been implemented.